File: java_api/src/main/java/utils/ApiUtils.java
package utils;

import org.springframework.util.MultiValueMap;

public class ApiUtils {

  public static Object getParamString(Constants aParameter, MultiValueMap<String, String> aAllParameters){
    if(aAllParameters.getFirst(aParameter.getKey()) != null)
      return aAllParameters.getFirst(aParameter.getKey());
    else
      return aParameter.getDefault();
  }
}

File: kotlin_api/src/main/kotlin/utils/ApiUtils.kt
package utils

import org.springframework.util.MultiValueMap

class ApiUtils {
    companion object {
        fun getParam(aArg: Constants, aParam: MultiValueMap<String, String>): Any? {
            if(!aParam.getFirst(aArg.getKey()).isNullOrEmpty())
                return aParam.getFirst(aArg.getKey())
            else
                return aArg.getDefault()
        }
    }
}